web3/middleware/signing.py

```diff
@@ -0,0 +1,92 @@
+from eth_account.local import (
+    LocalAccount,
+)
+from eth_keys.datatypes import (
+    PrivateKey,
+)
+from eth_utils import (
+    is_same_address,
+)
+from toolz import (
```

Comment:
have a look at #809. This should be change to `from web3.utils.toolz import compose`